math/mlp/src/TMultiLayerPerceptron.cxx

```diff
@@ -458,9 +466,15 @@ TMultiLayerPerceptron::TMultiLayerPerceptron(const char * layout, TTree * data,
    fData = data;
    fCurrentTree = -1;
    fCurrentTreeWeight = 1;
-   fTraining = new TEventList(Form("fTrainingList_%zu",(size_t)this));
+   {
+      RGDirectoryDetacher raii;
```

Comment:
I believe this should be done via
```suggestion
      TDirectory::TContext ctxt;
```

Reply:
absolutely. Sorry for forgetting about that.